lib/phoenix_live_view.ex

```diff
@@ -875,28 +875,41 @@ defmodule Phoenix.LiveView do
   The update function receives the current key's value and
   returns the updated value. Raises if the key does not exist.
 
+  The update function may also be of arity 2, in which case it receives
+  the current key's value as the first argument and the current assigns
+  as the second argument. Raises if the key does not exist.
+
   ## Examples
 
       iex> update(socket, :count, fn count -> count + 1 end)
       iex> update(socket, :count, &(&1 + 1))
+      iex> update(socket, :max_users_this_session, fn current_max, %{users: users} -> max(current_max, length(users)) end)
   """
   def update(socket_or_assigns, key, fun)
 
+  def update(%Socket{assigns: assigns} = socket, key, fun) when is_function(fun, 2) do
+    update(socket, key, &fun.(&1, assigns))
```

Comment:
Wait a minute :)
Here `assigns` is passing reserved keys like `:__changed__`, `:flash` and others which IMO should be stripped out as those keys are private.

You could call https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.Helpers.html#assigns_to_attributes/2 to clean up `assigns`

Reply:
No, it is fine to preserve those keys. `assigns` will always have those keys anyway, anywhere, anytime. :)

Reply:
All right, even simpler 👍🏻